### DISCUSSION TITLE
[Appium WebdriverIO][CucumberJS] - Is it possible to run the WebdriverIO using "npx cucumber-js" ?

### DISCUSSION THREAD

**Discussion by tamaniniandre** [webdriverio/webdriverio#12426] [Q&A] [answered]
Hello everyone! 

Well, I'm testing an APP built in Node.js, and I was having some blockers due to the XCUitest not showing all the elements in the App Source tree

More context: 
https://discuss.appium.io/t/error-in-appium-desktop-refreshing-source-after-adding-snapshotmaxdepth-greater-than-62/34058
https://discuss.appium.io/t/we-are-experiencing-max-deep-layer-problem/40399
https://github.com/appium/appium/issues/19156

So I decided to build a hybrid mobile automation with WebDriverIO and Detox (since the login flow only WebDriverIO/Appium can handle due to being a third party service).

I'm using cucumber.js to run the tests, and I would like to check the possibility to run both frameworks using the npx cucumber-js command. Is there some documentation that I can follow to do that?

For now, they run separately, basically the Appium is creating a new user and doing the login, then Detox does the rest, and I'm using the no-reset plan to get the authenticated session.

Accept new ideas.

Thanks so much,
André Tamanini

**Comment by christian-bromann** [comment, marked as answer]
Hey @tamaniniandre you can always spin up a WebdriverIO session in [standalone mode](https://webdriver.io/docs/setuptypes#standalone-mode) but this will require you to set up Appium yourself as you can't use the Appium service outside of the test runner.

**Reply by tamaniniandre** [reply]
got it, thanks so much!

**Comment by tamaniniandre** [comment]
hey @christian-bromann ,

by the way, do you have please an example for capabilities for Appium as example?
I'm trying this:

`
    {
        platformName: 'iOS',
        maxInstances: 1,
        'appium:deviceName': 'iPhone 13 Pro',
        'appium:platformVersion': '15.5',
        'appium:orientation': 'PORTRAIT',
        'appium:automationName': 'XCUITest',
        'appium:app': `${process.cwd()}/app/ios/tyb.app`,
        'appium:noReset': false,
        'appium:newCommandTimeout': 60000,
        'appium:settings[snapshotMaxDepth]': 60,
        'appium:settings[customSnapshotTimeout]': 600,
    },
];`

but I'm getting the following error:

`If you like to run a local browser session make sure to pick from one of the following browser names: chrome,googlechrome,chromium,chromium-browser,firefox,ff,mozilla,mozilla firefox,edge,microsoftedge,msedge,safari,safari technology preview
    at Runtime.runTestRunHooks (/Users/ubimnids/workspace/tyb-qa/node_modules/@cucumber/cucumber/src/runtime/run_test_run_hooks.ts:32:19)
    at async Runtime.start (/Users/ubimnids/workspace/tyb-qa/node_modules/@cucumber/cucumber/src/runtime/index.ts:103:5)
    at async runCucumber (/Users/ubimnids/workspace/tyb-qa/node_modules/@cucumber/cucumber/src/api/run_cucumber.ts:125:19)
    at async Cli.run (/Users/ubimnids/workspace/tyb-qa/node_modules/@cucumber/cucumber/src/cli/index.ts:79:25)
    at async Object.run [as default] (/Users/ubimnids/workspace/tyb-qa/node_modules/@cucumber/cucumber/src/cli/run.ts:33:14)
caused by: Error: No "browserName" defined in capabilities nor hostname or port found!
If you like to run a local browser session make sure to pick from one of the following browser names: chrome,googlechrome,chromium,chromium-browser,firefox,ff,mozilla,mozilla firefox,edge,microsoftedge,msedge,safari,safari technology preview`
